fastseq/optimizer/transformers/beam_search_optimizer.py

```diff
@@ -115,915 +179,1018 @@ def _update_beam_size(self, num_beams):
             logger.debug("num_beams has been updated to {}".format(num_beams))
             return
 
+        if isinstance(self, ProphetNetForConditionalGeneration):
+            for layer in self.prophetnet.decoder.layers:
+                #layer.self_attn.num_beams = num_beams
+                layer.cross_attn.num_beams = num_beams
+            logger.debug("num_beams has been updated to {}".format(num_beams))
+            return
+
         logger.debug(
             "The num_beams optimization in self_attn and encoder_decoder_attn "
             "does not support {} yet.".format(self.__class__))
 
 
+    def _get_logits_processor(
```

Comment:
It replaces: https://github.com/huggingface/transformers/blob/69e16abf98c94b8a6d2cf7d60ca36f13e4fbee58/src/transformers/generation_utils.py#L559. It should be getting replaced when GenerationMixinV2 replaces GenerationMixin

Reply:
Is `@replace(...)` needed here?

Reply:
I don't think so, since it is a member of GenerationMixin which is already replaced here: https://github.com/microsoft/fastseq/blob/c3c98cbc10e13f7bcb7c456b3531d7438fafa948/fastseq/optimizer/transformers/beam_search_optimizer.py#L133